plugins/elanfp/fu-elanfp-device.c

```diff
@@ -40,40 +42,75 @@ fu_elanfp_iap_send_command(FuElanfpDevice *self,
 			   guint8 request,
 			   const guint8 *buf,
 			   gsize bufsz,
+			   gsize rspsz,
 			   GError **error)
 {
 	GUsbDevice *usb_device = fu_usb_device_get_dev(FU_USB_DEVICE(self));
+	guint16 pid = (guint)fu_usb_device_get_pid(FU_USB_DEVICE(self));
 	gsize actual = 0;
-	guint8 buftmp[61] = {request, 0};
+	gsize sendsz = bufsz + 1;
+	guint8 start_index = 0x01;
+	guint8 buftmp[64] = {request, 0};
+	gboolean is_new_protocol = FALSE;
+
+
+	if ((pid != 0x0C7E) && (pid != 0x0C82)) {
+					is_new_protocol = TRUE;
+	}
 
 	if (buf != NULL) {
+		if (is_new_protocol) {
+					buftmp[0] = TAG_SEND_COMMAND;
+					buftmp[1] = rspsz;
+					buftmp[2] = bufsz + 1;
+					buftmp[3] = request;
+					start_index = 0x04;
+					sendsz = bufsz + 1 + 3;
+		}
 		if (!fu_memcpy_safe(buftmp,
 				    sizeof(buftmp),
-				    0x1, /* dst */
+				    start_index, /* dst */
 				    buf,
 				    bufsz,
 				    0x0, /* src */
 				    bufsz,
 				    error))
 			return FALSE;
 	}
-	if (!g_usb_device_control_transfer(usb_device,
+
+	if (is_new_protocol) {
+		if (!g_usb_device_bulk_transfer(usb_device,
+					ELAN_EP_CMD_OUT,
+					buftmp,
+					sendsz,
+					&actual,
+					BULK_SEND_TIMEOUT_MS,
+					NULL,
+					error)) {
+		g_prefix_error(error, "failed to send command (bulk): ");
+		return FALSE;
+		}
+	}
+	else {
```

Comment:
intending is weird, you need to use contrib/setup so that clang-format can auto-fix this for you.

Reply:
I ran the pre-commit run --hook-stage commit --all-files at local side and got all pass results. But this commit id e1fc9869b5b22a463de2ecc01a333aecad900d44 still got fail result on the item clang-format  on the web side. How to get relative fail messages to fix it.

Reply:
if you squash the patches together (e.g. `git rebase main -i`) the CI should be happy.